src/pages/singleproduct/SingleProduct.jsx

```diff
@@ -0,0 +1,134 @@
+import "./singleproduct.css";
+import { useParams, useNavigate } from "react-router-dom";
+import { useProduct } from "context/product-context";
+import { FastDelivery, InStock, ReturnPolicy } from "assets/icons";
+import { Toast } from "component";
+import { useCart } from "context/cart-context";
+import { useToast } from "context/toast-context";
+import { Link } from "react-router-dom";
+import ReactImageMagnify from "react-image-magnify";
+
+export const SingleProduct = () => {
+  const { productList } = useProduct();
+  const param = useParams();
+  const navigate = useNavigate();
+  let productId = param.productId;
+  const getProduct = (id) => productList.find(({ _id }) => id === _id);
+  const product = getProduct(productId, 10);
+  const {
+    _id,
+    description,
+    image,
+    name,
+    price,
+    ratings,
+    discounted,
+    original,
+  } = product;
+
+  const {
+    addToCartHandler,
+    cartState: { cartData },
+  } = useCart();
+
+  const isInCartList = (id) => cartData.find(({ _id }) => _id == id);
+  const clickHandler = (product) => {
+    addToCartHandler(product);
+  };
+
+  return (
+    <>
+      <Toast />
+      <div className="shadow-box product_wrapper">
+        <div className="image_wrapper ">
+          <ReactImageMagnify
+            {...{
+              smallImage: {
+                alt: { name },
+                src: image,
+                width: 450,
+                height: 450,
+              },
+
+              largeImage: {
+                src: image,
+                width: 1500,
+                height: 1800,
+              },
+              enlargedImageContainerDimensions: {
+                width: "180%",
+                height: "100%",
+              },
+            }}
+          />
+        </div>
+        <div className="detail_wrapper vertical-direction">
+          <h1 className="text-primary">{name}</h1>
+          <p className="mb-lg">{description}</p>
+          <div className="mb-lg text-primary">
+            {[...Array(5)].map((a, ind) => {
+              return (
+                <span key={ind}>
+                  <i
+                    className={
+                      ratings >= ind + 1
+                        ? "fa fa-star fa-lg"
+                        : "far fa-star fa-lg"
+                    }
+                  ></i>
+                </span>
```

Comment:
What is `[...Array(5)]` here?

Reply:
So basically I am creating an array having 5 elements and using its index checking if rating is greater than it or not and setting the class according to that